edge-apps/clock/gulpfile.js

```diff
@@ -0,0 +1,59 @@
+const { src, dest, series } = require('gulp')
+const babel = require('gulp-babel')
+const base64 = require('gulp-base64')
+const del = require('del')
+const inlineSource = require('gulp-inline-source')
+const replace = require('gulp-replace')
+const uglifyJS = require('gulp-uglify')
+const uglifyCSS = require('gulp-uglifycss')
+
+function buildJS () {
+  return src('build/static/js/*.js')
+    .pipe(babel())
+    .pipe(uglifyJS())
+    .pipe(dest('build/static/js'))
+}
+
+function buildCSS () {
+  return src('build/static/styles/*.css')
+    .pipe(uglifyCSS())
+    .pipe(dest('build/static/styles'))
+}
+
+function cleanDist () {
+  return del(['dist/**/*', '!dist'])
+}
+
+function cleanBuild () {
+  return del('build')
+}
+
+function copyAssets () {
+  return src(['src/**'])
+    .pipe(dest('build'))
+}
+
+function replaceKeys () {
+  return src(['build/index.html'])
+    .pipe(replace('GA_API_KEY', process.env.GA_API_KEY))
```

Comment:
I don't think this logic is sufficient. Sure, it's fine if it's defined, but if it is undefined, we are trying to load GA with the API key `GA_API_KEY`. We probably need to put it behind and if statement.